core/model/modx/processors/resource/update.class.php

```diff
@@ -452,6 +452,18 @@ public function checkForUnPublishOnSiteStart() {
      * @return boolean
      */
     public function checkDeletedStatus() {
+	    if (1 == $this->getProperty('deleted', null)) {
+	    	if ($this->modx->getOption('site_start') == $this->getProperty('id')) {
```

Comment:
@rtripault $this->modx->getOption('site_start') should get a context over a system setting too. Hope I am right with that.